docs/AA--Javascript-in-EE/asset-caching-by-filename.md

```diff
@@ -6,30 +6,76 @@ We also briefly considered using `filemtime` as the "version" generator for our
 
 So we've decided to [switch to filenames containing a hash](https://github.com/eventespresso/event-espresso-core/pull/287) generated from the content of the files.  This means if the content of a built file does not change, then the filename remains the same, if the content of a built bundle file changes, then the filename will as well.
 
-Switching to this new process requires some tweaks to how our assets are registered with `wp_register_script` however.
+Switching to this new process requires some tweaks to where assets are located, and how they are registered with `wp_register_script`.
+
+## Where Javascript and CSS Assets Should be Located
+
+In Event Espresso core, new Javascript and CSS files should be located somewhere under `assets/src` (previously they were placed in `core/templates/global_assets`, or beside the PHP files they corresponded to.)
+Put the files in whatever folder/subfolder is logical, like `assets/src/empire/stardestroyer.js'.
+`
+Then add an new entry to `webpack.common.js`' `config` const. Use whatever string you like for the entry slug/chunk name.
+Like the following
+
+```javascript
+...
+{
+		entry: {
+			'eventespresso-core-star-destroyer-chunkname' /* <!-- entry slug/chunk name! */: [
+				assets + 'empire/stardestroyer.js',
+			],
+		},
+		output: {
+			filename: '[name].[chunkhash].dist.js',
+			path: path.resolve( __dirname, 'assets/dist' ),
+		},
+		module: moduleConfigWithJsAndCssRules,
+		watchOptions: {
+			poll: 1000,
+		},
+	},
+...
+```
+You've now informed webpack where the new assets are located, and what their chunk name is.
+
+Don't forget to run the build process with `npm run watch` to build the files. 
```

Comment:
maybe:

> Don't forget to run the build process with `npm run watch` for development sites or `npm run build` for production sites.

Reply:
Hmmm but [our docs elsewhere say](https://github.com/eventespresso/event-espresso-core/blob/master/docs/AA--Javascript-in-EE/build-process.md#automation) we basically shouldn't ever use `npm run build` unless we're testing that it works, seeing how the build process automatically does that when there's a release. And I cringe to think someone might attempt running `npm run build` on a live site...

Reply:
fair enuf... I was considering someone that may be customizing a site but I guess we don't want to encourage that.

Reply:
I suppose there is a use-case for it before the site goes live. But ya, that would fall into honey badgering territory.